confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java

```diff
@@ -362,4 +384,70 @@ public void clear() throws AuthException {
     }
     authorizer.reset();
   }
+
+  /**
+   * When the permission information of a user or role is changed, the permission cache information
+   * of all datanode is initialized, and only the affected cache is initialized.
+   *
+   * @param env
+   * @param username
+   * @param roleName
+   * @throws IOException
+   * @throws TException
+   */
+  public void invalidateCache(ConfigNodeProcedureEnv env, String username, String roleName)
+      throws IOException, TException {
+    List<TDataNodeInfo> allDataNodes =
+        env.getConfigManager().getNodeManager().getOnlineDataNodes(-1);
+    TInvalidatePermissionCacheReq req = new TInvalidatePermissionCacheReq();
+    req.setUsername(username);
+    req.setRoleName(roleName);
+    for (TDataNodeInfo dataNodeInfo : allDataNodes) {
+      env.getDataNodeClient(dataNodeInfo.getLocation()).invalidatePermissionCache(req);
+    }
+  }
+
+  /**
+   * Save the user's permission information,Bring back the datanode for caching
+   *
+   * @param username The username of the user that needs to be cached
+   * @return TPermissionInfoResp
+   * @throws AuthException
+   */
+  public TPermissionInfoResp getUserPermissionInfo(String username) throws AuthException {
+    TPermissionInfoResp result = new TPermissionInfoResp();
+    TUserResp tUserResp = new TUserResp();
+    TRoleResp tRoleResp = new TRoleResp();
+    Map<String, TRoleResp> tRoleRespMap = new HashMap();
+    List<String> userPrivilegeList = new ArrayList<>();
+    List<String> rolePrivilegeList = new ArrayList<>();
+
+    // User permission information
+    User user = authorizer.getUser(username);
+    for (PathPrivilege pathPrivilege : user.getPrivilegeList()) {
+      userPrivilegeList.add(pathPrivilege.getPath());
+      String privilegeIdList = pathPrivilege.getPrivileges().toString();
+      userPrivilegeList.add(privilegeIdList.substring(1, privilegeIdList.length() - 1));
```

Comment:
Please check if `privilegeIdList` can be null or shorter than 1.

Reply:
fix~